File: Array | Sets/contain_duplicate/js/contain_duplicate.js
console.log('Contain Duplicates')

const solution = {
    containDuplicate(nums=[]) {
        const set = new Set();

        for ( let num of nums ) {
            if ( ! set.has(num) ) {
                set.add(num)
            } else {
                return true;
            }
        }

        return false;
    },

    init: function () {
        this.containDuplicate([1,2,3,1])
    }
}

solution.init();

File: Array | Sets/contain_duplicate/py/contain_duplicate.py
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        set = {}
        for num in nums:
            if not set.get(num):
                set[num] = 1
                # set.update({num:1})
            else:
                return True
        return False

sol = Solution()
print(sol.containsDuplicate([1,2,3,1]))
